### DISCUSSION TITLE
How to integrate custom server with next internationalized routing feature

### DISCUSSION THREAD

**Discussion by aNyMoRe0505** [vercel/next.js#25667] [Help] [answered]
Hello, currently I have a next application with custom server and i18n

since `v10.0.0`,  next.js has built-in support for internationalized routing

so I refactor my code and update my package (i18n included: https://github.com/isaachinman/next-i18next)

but I found seems currently has no way to integrate custom server and internationalized routing feature

here is [sample code](https://github.com/aNyMoRe0505/next-custom-server-i18n)

(npm i && npm run dev)

when I landing to a page, I can't get the locale info by router

Is there has any way to integrate custom server and internationalized routing feature ?

thanks !

**Comment by timtnleeProject** [comment]
@aNyMoRe0505 
Will it work if you put the next config with internationalized routing correctly in `server.js`?

```javascript
const next = require('next')
const app = next({
  conf: {
    i18n: {
      locales: ['en-us', 'zh-hk', 'fr'],
      defaultLocale: 'en-us',
    }
  }
})
```

**Reply by aNyMoRe0505** [reply]
@timtnleeProject 

thank your suggestion, I tried but situation remain same

It seems that It will override the original config, so I think we don't need to put the next config

**Comment by aNyMoRe0505** [comment, marked as answer]
for those who facing same problem you can check this disscusion: https://github.com/vercel/next.js/discussions/24718

and the issue here: https://github.com/vercel/next.js/issues/20761
